connectors/vault/vaultconn-bl/vault_reader.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package vaultconnbl
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	vaultutils "github.com/ibm/the-mesh-for-data/connectors/vault/vault_utils"
+	pb "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf"
+)
+
+type Server struct {
+}
+
+func (s *Server) GetCredentialsInfo(in *pb.DatasetCredentialsRequest, vault vaultutils.VaultConnection, vaultPathKey string) (*pb.DatasetCredentials, error) {
+	log.Println("Vault connector: GetCredentialsInfo")
+	log.Printf("Vault connector: GetCredentialsInfo, in = %v\n", in)
+
+	// secretaddr := url.QueryEscape(in.DatasetId)
+	secretaddr := strings.ReplaceAll(in.DatasetId, " ", "")
+
+	readCredentials, err := vault.GetFromVault(vaultPathKey, secretaddr)
+	if err != nil {
+		log.Printf("Error in vaultConnector, got error from the vault: %v\n", err.Error())
+		return nil, fmt.Errorf("error in retrieving the secret from vault(key = %s): %v", secretaddr, err)
+	}
+	log.Println("Read credentials from vault: " + readCredentials)
```

Comment:
Please do not log credentials

Reply:
sometimes we may need to check for debugging purposes if the correct credentials are being retrieved. so can we have a flag based approach here. if the flag is set log the credentials otherwise dont log it. is this ok? @froesef

Reply:
A flag also introduces potential risks. Credentials should never be logged. If you need something for debugging or testing there are probably better ways to test this. 

I think in this setup you have a vault server that is mocked and a GetCredentials method that returns the credentials that you can then check using assert checks in the test. If you're worried about different things that you can put into vault you can write tests about what happens when e.g. wrong input get's put into vault.

But any logging of credentials is a potential security issue.

Reply:
ok sure. So writing tests is the only approach as you have mentioned. So there is no way we can at runtime verify if the correct credentials have been got. I will remove the logging. Thanks.